src/lib/server/orders.ts

```diff
@@ -626,9 +627,11 @@ export async function createOrder(
 			.filter((d) => d._id !== null)
 			.map((d) => [
 				d._id,
-				wholeDiscount !== undefined && wholeDiscount > d.discountPercent
+				wholeDiscount !== undefined &&
+				d.discountPercent !== undefined &&
+				wholeDiscount > d.discountPercent
 					? wholeDiscount
-					: d.discountPercent
+					: d.discountPercent ?? 0
```

Comment:
This is still not correct.

Delete all the discounts you have.

Create just a whole discount of 50%.

Does the product still get the discount or not? (before / after this PR)